client/src/views/all-games/components/SignupForm.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC, ReactElement, useRef, FormEvent } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Game } from 'shared/typings/models/game';
+import { Signup } from 'client/typings/user.typings';
+import { RootState } from 'client/typings/redux.typings';
+import {
+  submitSignup,
+  submitSelectedGames,
+} from 'client/views/signup/signupActions';
+
+interface Props {
+  game: Game;
+  startTime: string;
+}
+
+export const SignupForm: FC<Props> = (props: Props): ReactElement => {
+  const { game, startTime } = props;
+
+  const dispatch = useDispatch();
+  const selectedGames: readonly Signup[] = useSelector(
+    (state: RootState) => state.myGames.signedGames
+  );
+  const username: string = useSelector(
+    (state: RootState) => state.login.username
+  );
+  const priorityRef = useRef<HTMLSelectElement>(null);
+
+  const handleSignup = async (event: FormEvent): Promise<void> => {
+    event.preventDefault();
+    if (!priorityRef.current) {
+      return;
+    }
+
+    const newGame = [
+      {
+        gameDetails: game,
+        priority: parseInt(priorityRef.current.value, 10),
+        time: game.startTime,
+      },
+    ];
+
+    const combined = selectedGames.concat(newGame);
+    dispatch(submitSelectedGames(combined));
+
+    const signupData = {
+      username,
+      selectedGames: combined,
+      signupTime: game.startTime,
+    };
+
+    try {
+      await dispatch(submitSignup(signupData));
+    } catch (error) {
+      switch (error.code) {
+        case 40:
```

Comment:
Palauttaako tää rajapinta jossakin tapauksessa `code: 40` vai pitäskö olla `41`?

Oon koittanut dokumentoida noita tiedostoon `server/docs/error-codes.txt` ja äkkiseltään näyttäisi, että tuossa halutaan kuunnella virhettä `41`.

![image](https://user-images.githubusercontent.com/1327412/113588949-6fec6e00-9639-11eb-8045-59ddec395c4c.png)

Noitakin voisi koittaa tyypittää niin ei tarvii mennä puhtaalla käsipelillä.

Reply:
Oliskohan `readme.md` parempi paikka noille virhekoodeille? Tai ainakin voisi linkata readme-tiedostoon, että mistä ne löytyy :D

Reply:
Tarkistin endpointin ja pitäisi tosiaan olla 41. Voisi kyllä olla hyvä idea koittaa tyypittää näitä 🤔